src/solidlsp/language_servers/scala_language_server.py

```diff
@@ -0,0 +1,267 @@
+            config.code_language.value,
+            solidlsp_settings,
+        )
+        self.server_ready = threading.Event()
+        self.initialize_searcher_command_available = threading.Event()
+
+    @override
+    def is_ignored_dirname(self, dirname: str) -> bool:
+        return super().is_ignored_dirname(dirname) or dirname in [
+            ".bloop",
+            ".metals",
+            "target",
+        ]
+
+    @classmethod
+    def _setup_runtime_dependencies(
+        cls, logger: LanguageServerLogger, config: LanguageServerConfig, solidlsp_settings: SolidLSPSettings
+    ) -> list[str]:
+        """
+        Setup runtime dependencies for Scala Language Server and return the command to start the server.
+        """
+        assert shutil.which("java") is not None, "JDK is not installed or not in PATH."
+
+        metals_home = os.path.join(cls.ls_resources_dir(solidlsp_settings), "metals-lsp")
+        os.makedirs(metals_home, exist_ok=True)
+        metals_executable = os.path.join(metals_home, "metals")
+        coursier_command_path = shutil.which("coursier")
+        cs_command_path = shutil.which("cs")
+
+        if not coursier_command_path:
+            raise RuntimeError("'coursier' not found isn't in PATH. Please install coursier try again.")
+
+        if not cs_command_path:
+            logger.log("'cs' command not found. Trying to install it using 'coursier'.", logging.INFO)
+            try:
+                logger.log("Running 'coursier setup --yes' to install 'cs'...", logging.INFO)
+                subprocess.run([coursier_command_path, "setup", "--yes"], check=True, capture_output=True, text=True)
+            except subprocess.CalledProcessError as e:
+                raise RuntimeError(f"Failed to set up 'cs' command with 'coursier setup'. Stderr: {e.stderr}")
+
+            cs_command_path = shutil.which("cs")
+            if not cs_command_path:
+                raise RuntimeError("'cs' command not found after running 'coursier setup'. Please check your PATH or install it manually.")
+            logger.log("'cs' command installed successfully.", logging.INFO)
+
+        if not os.path.exists(metals_executable):
+            logger.log(f"metals executable not found at {metals_executable}, bootstrapping...", logging.INFO)
+            artifact = "org.scalameta:metals_2.13:1.6.2"
+            cmd = [
+                cs_command_path,
+                "bootstrap",
+                "--java-opt",
+                "-XX:+UseG1GC",
+                "--java-opt",
+                "-XX:+UseStringDeduplication",
+                "--java-opt",
+                "-Xss4m",
+                "--java-opt",
+                "-Xms100m",
+                "--java-opt",
+                "-Dmetals.client=Serena",
+                artifact,
+                "-o",
+                metals_executable,
+                "-f",
+            ]
+            logger.logger.info("Bootstrapping metals...")
```

Comment:
nitpick - pls use logger.log so it's the same pattern everywhere

Reply:
https://github.com/oraios/serena/pull/528/commits/268562984d52adbaa3bbb92486f2e69cf20e2b04

Reply:
https://github.com/oraios/serena/pull/528/commits/cd1b2ba8547510a88e55cf7fd793991a9103fdef